Adjusts default.js to be in line with docs, and adds package fields.

qubyte opened the pull request:
I'm using ES2015 modules (particularly with rollup) a lot for my personal projects these days. zen-observable almost works, but there were a couple of snags which this PR addresses. The docs suggest what this module can be required in using:

```javascript
import Observable from "zen-observable";
```

but with default.js and the package file as they are in master this line needs to be:

```javascript
import { Observable } from "zen-observable/default.js";
```

This PR adjusts default.js and adds a `module` field to the package file for rollup to pick up (and I'm told webpack too). [This field is to ES2015 modules as the `main` field is to Node style CommonJS modules](https://github.com/rollup/rollup/wiki/pkg.module). These changes bring the behaviour of this module into line with the docs without affecting internal modules.

zenparsing commented:
@qubyte I don't quite understand why

```js
import Observable from 'zen-observable';
```

isn't working for you. Is rollup or webpack not able to find the module?

In general, I'd rather only support linking to "index.js" right now.

qubyte commented:
The big assumption I'm making throughout is that default.js is intended for ES module usage. This makes sense since UMD modules and ES modules are incompatible, so they must live is separate files.

That being the case, there are two issues which this PR addresses:

The first is that `default.js` doesn't have a default export, so `import Observable ...` doesn't work, but `import { Observable } ...` does. I've adjusted `default.js` to fix this.

The second problem is that rollup doesn't know to look at default.js. Adding a CJS plugin adds a lot of fluff to the build and sort of goes against the point of using an ES module loader / bundler. The quick solution to that problem is to do `from "zen-observable/default.js"`, but that's a bit clunky. The `module` field I've added to the package file is only used by ES module loaders (see link in description for more info on that), so it won't affect consumers which use UMD modules (which naturally look at the `main` field or default to index.js).

qubyte commented:
I'm worried that my explanations have not been helpful. Would it be useful to provide a little repo which demonstrates the issue?

zenparsing commented:
@qubyte Apologies for the delay in responding.

Actually, index.js is the intended entry point for both `require` and `import`. The `default.js` file isn't needed and will likely be removed.

qubyte commented:
I think that's the issue. UMD and ES2015 module systems are incompatible. For an ES2015 module loader to consume index.js (including emerging support in browsers) the CJS needs to be transpiled out (think browserify etc.) or a plugin used (not clear how this work work for a browser). Large and complex module loaders like webpack are capable of this, whereas vanilla ES2015 module loaders such as the browser won't know what to do.

Given that support and use of ES2015 modules is set to grow, I reckon I'll only be the first to bring up the issue. Other modules are beginning to be shipped as ES2015, with a rollup compiled UMD module for legacy systems. One notable example is [threejs][2]. [Most][3] [of][4] [my][5] [own][6] libraries ship like this too (I use rollup on the `prepublish` hook).

P.S.

I put three tiny demos into [this repo][1]. It demonstrates the problem as is, the fixes in the PR, and also the workaround I currently use.

Regardless of the outcome of this PR, thank you for this module. Everything is so much nicer with observables!

[1]: https://github.com/qubyte/observable-issue-demo
[2]: https://github.com/mrdoob/three.js/
[3]: https://www.npmjs.com/package/es2015-deferred
[4]: https://www.npmjs.com/package/mixomatic
[5]: https://www.npmjs.com/package/vertebrate-event-emitter
[6]: https://www.npmjs.com/package/navbar

zenparsing commented:
I see now, Rollup only supports ES modules out of the box. Thanks!
